src/Compilers/CSharp/Portable/BoundTree/UnboundLambda.cs

```diff
@@ -162,6 +173,11 @@ protected override BoundExpression VisitExpressionWithoutStackGuard(BoundExpress
 
             public override BoundNode VisitReturnStatement(BoundReturnStatement node)
             {
+                if (node.RefKind != RefKind.None)
+                {
+                    refKind = node.RefKind;
```

Comment:
It would be good to have a comment about how are we dealing with ambiguities, i.e. refness doesn't agree across multiple returns.